src/compounds/information-panel/src/index.tsx

```diff
@@ -0,0 +1,283 @@
+  rows
+}) => (
+  <table
+    sx={{
+      border: 'none',
+      borderCollapse: 'collapse',
+      overflow: 'scroll',
+      margin: 0,
+      width: '100%'
+    }}
+  >
+    <tbody>
+      {rows.map((row: any[], i: number) => {
+        return (
+          <tr
+            sx={{ variant: 'compounds.information-panel.table.trSx' }}
+            key={i}
+          >
+            {row.map((cell: any, j) => {
+              return cell.type === 'th' ? (
+                <th
+                  key={j}
+                  colSpan={cell.colspan}
+                  sx={{ variant: 'compounds.information-panel.table.tdSx' }}
+                >
+                  {cell.value}
+                </th>
+              ) : (
+                <td
+                  colSpan={cell.colspan}
+                  sx={{ variant: 'compounds.information-panel.table.tdSx' }}
+                  key={j}
+                >
+                  {cell.value}
+                </td>
+              )
+            })}
+          </tr>
+        )
+      })}
+    </tbody>
+  </table>
+)
+
+interface MoreInformationBlockProps
+  extends React.HTMLAttributes<HTMLDivElement> {
+  title: string
+}
+
+export const MoreInformationBlock: React.FC<MoreInformationBlockProps> = ({
+  title,
+  children
+}) => (
+  <div
+    sx={{
+      background: '#fff',
+      color: '#333',
+      alignItems: 'center',
+      fontSize: '14px',
+      boxSizing: 'border-box',
+      display: 'block',
+      width: ['auto', '50%'],
+      marginTop: 'sm',
+      marginBottom: [0, 'sm'],
+      marginLeft: 'sm',
+      marginRight: 'sm',
+      ':nth-of-type(even)': {
+        marginLeft: ['sm', '0']
+      }
+    }}
+  >
+    <div
+      sx={{
+        padding: '15px 20px 20px',
+        lineHeight: ['1.618em', 'inherit']
+      }}
+    >
+      <div sx={{ variant: 'compounds.information-panel.block.title' }}>
+        {title}
+      </div>
+
+      <div sx={{ background: '#fff', overflow: 'auto' }}>{children}</div>
+    </div>
+  </div>
+)
+
+interface EligibilityProps extends React.HTMLAttributes<HTMLDivElement> {
+  moreInformationPanel: React.ReactNode[]
+  moreInformationButton?: React.ReactNode
+  moreInformationLabel?: string
+}
+
+export const Eligibility: React.FC<EligibilityProps> = ({
+  moreInformationPanel,
+  moreInformationButton,
+  moreInformationLabel = 'More information'
+}) => {
+  const [open, setOpen] = React.useState(false)
+
+  return (
+    <div sx={{ background: '#f2f3f4' }}>
+      <div
+        sx={{
+          display: 'block',
+          margin: 'auto',
+          height: open ? 'auto' : '0px',
+          transition: 'height .4s ease-in-out',
+          overflow: 'hidden'
+        }}
+      >
+        <div
+          sx={{
+            display: 'flex',
+            flexDirection: ['column', 'row'],
+            alignItems: 'stretch',
+            background: '#f2f3f4',
+            justifyContent: 'center'
+          }}
+        >
+          {moreInformationPanel}
```

Comment:
why don't we instantiate the components inside the main component but get them as props?

Reply:
It would make these more reusable if this parrt of the logic were in the product table and it was just the three blocks in this component

Reply:
In this case, the content of the panel is dynamic. It changes depending on the product's data. One product might have a list and another one might have a list and a table in its more information panel. We give an array of components  as a prop depending on the products data and eligibility renders them.

Reply:
> It would make these more reusable if this parrt of the logic were in the product table and it was just the three blocks in this component

We can still import text, table, or list separately. I thought that if I left the eligibility part inside the legacy product table, I would need implement the similar thing also into the product table and we would have it in two different places.